rust/src/go/nmstate/nmstate.go

```diff
@@ -0,0 +1,111 @@
+package nmstate
+
+// #cgo CFLAGS: -g -Wall
+// #cgo LDFLAGS: -lnmstate
+// #include <nmstate.h>
+// #include <stdlib.h>
+import "C"
+import "fmt"
+
+type Flags uint
+
+const None Flags = 0
+
+const (
+	KernelOnly Flags = 1 << iota
+	NoVerify
+	IncludeStatusData
+	IncludeSecrets
+	NoCommit
+)
+
+func RetrieveNetState(flags Flags, logs io.Writer) (string, error) {
+	var (
+		state    *C.char
+		log      *C.char
+		err_kind *C.char
+		err_msg  *C.char
+	)
+	rc := C.nmstate_net_state_retrieve(C.uint(flags), &state, &log, &err_kind, &err_msg)
+
+	defer func() {
+		C.nmstate_net_state_free(state)
+		C.nmstate_err_msg_free(err_msg)
+		C.nmstate_err_kind_free(err_kind)
+		C.nmstate_log_free(log)
+	}()
+	io.WriteString(logs, C.GoString(log))
```

Comment:
You have handler `WriteString` error https://pkg.go.dev/io#WriteString

```golang
if err := io.WriteString(logs, C.GoString(log)); err != nil {
    return "", fmt.Errorf("failed wiring logs: %v", err)
}
```

Reply:
Done, thank you! :-)